plugins/inputs/t128_graphql/t128_graphql.go

```diff
@@ -0,0 +1,271 @@
+package t128_graphql
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/Jeffail/gabs"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	// DefaultRequestTimeout is the request timeout if none is configured
+	DefaultRequestTimeout = time.Second * 5
+)
+
+// T128GraphQL is an input for metrics of a 128T router instance
+type T128GraphQL struct {
+	CollectorName string            `toml:"collector_name"`
+	BaseURL       string            `toml:"base_url"`
+	UnixSocket    string            `toml:"unix_socket"`
+	EntryPoint    string            `toml:"entry_point"`
+	Fields        map[string]string `toml:"extract_fields"`
+	Tags          map[string]string `toml:"extract_tags"`
+	Timeout       internal.Duration `toml:"timeout"`
+
+	Query          string
+	JSONEntryPoint string
+	client         *http.Client
+}
+
+// SampleConfig returns the default configuration of the Input
+func (*T128GraphQL) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description on the Input
+func (*T128GraphQL) Description() string {
+	return "Make a 128T GraphQL query and return the data"
+}
+
+// Init sets up the input to be ready for action
+func (plugin *T128GraphQL) Init() error {
+	if plugin.CollectorName == "" {
+		return fmt.Errorf("collector_name is a required configuration field")
+	}
+
+	if plugin.BaseURL == "" {
+		return fmt.Errorf("base_url is a required configuration field")
+	}
+
+	if !strings.HasSuffix(plugin.BaseURL, "/") {
+		plugin.BaseURL += "/"
+	}
+
+	if plugin.EntryPoint == "" {
+		return fmt.Errorf("entry_point is a required configuration field")
+	}
+
+	if plugin.Fields == nil {
+		return fmt.Errorf("extract_fields is a required configuration field")
+	}
+
+	transport := http.DefaultTransport
+
+	if plugin.UnixSocket != "" {
+		transport = &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", plugin.UnixSocket)
+			},
+		}
+	}
+
+	plugin.Query = buildQuery(plugin.EntryPoint, plugin.Fields, plugin.Tags)
+	plugin.JSONEntryPoint = plugin.buildJSONPathFromEntryPoint()
+	plugin.client = &http.Client{Transport: transport, Timeout: plugin.Timeout.Duration}
+
+	return nil
+}
+
+// Gather takes in an accumulator and adds the metrics that the Input gathers
+func (plugin *T128GraphQL) Gather(acc telegraf.Accumulator) error {
+	timestamp := time.Now()
```

Comment:
If you just want to have the current timestamp on the data, I don't think you need to assign a specific timestamp. nbd though.